compose/material/material/src/skikoMain/kotlin/androidx/compose/material/AlertDialog.skiko.kt

```diff
@@ -0,0 +1,151 @@
+ */
+@Composable
+actual fun AlertDialog(
+    onDismissRequest: () -> Unit,
+    confirmButton: @Composable () -> Unit,
+    modifier: Modifier,
+    dismissButton: @Composable (() -> Unit)?,
+    title: @Composable (() -> Unit)?,
+    text: @Composable (() -> Unit)?,
+    shape: Shape,
+    backgroundColor: Color,
+    contentColor: Color,
+    properties: DialogProperties
+) {
+    AlertDialog(
+        onDismissRequest = onDismissRequest,
+        buttons = {
+            // TODO: move the modifiers to FlowRow when it supports a modifier parameter
+            Box(Modifier.fillMaxWidth().padding(horizontal = 8.dp, vertical = 2.dp)) {
+                AlertDialogFlowRow(
+                    mainAxisSpacing = 8.dp,
+                    crossAxisSpacing = 12.dp
+                ) {
+                    dismissButton?.invoke()
+                    confirmButton()
+                }
+            }
+        },
+        modifier = modifier,
+        title = title,
+        text = text,
+        shape = shape,
+        backgroundColor = backgroundColor,
+        contentColor = contentColor,
+        properties = properties
+    )
+}
+
+/**
+ * <a href="https://material.io/components/dialogs#alert-dialog" class="external" target="_blank">Material Design alert dialog</a>.
+ *
+ * Alert dialogs interrupt users with urgent information, details, or actions.
+ *
+ * ![Dialogs image](https://developer.android.com/images/reference/androidx/compose/material/dialogs.png)
+ *
+ * This function can be used to fully customize the button area, e.g. with:
+ *
+ * @sample androidx.compose.material.samples.CustomAlertDialogSample
+ *
+ * @param onDismissRequest Executes when the user tries to dismiss the Dialog by clicking outside
+ * or pressing the back button. This is not called when the dismiss button is clicked.
+ * @param buttons Function that emits the layout with the buttons.
+ * @param modifier Modifier to be applied to the layout of the dialog.
+ * @param title The title of the Dialog which should specify the purpose of the Dialog. The title
+ * is not mandatory, because there may be sufficient information inside the [text]. Provided text
+ * style will be [Typography.subtitle1].
+ * @param text The text which presents the details regarding the Dialog's purpose. Provided text
+ * style will be [Typography.body2].
+ * @param shape Defines the Dialog's shape.
+ * @param backgroundColor The background color of the dialog.
+ * @param contentColor The preferred content color provided by this dialog to its children.
+ * @param properties Typically platform specific properties to further configure the dialog.
+ */
+@Composable
+actual fun AlertDialog(
+    onDismissRequest: () -> Unit,
+    buttons: @Composable () -> Unit,
+    modifier: Modifier,
+    title: (@Composable () -> Unit)?,
+    text: @Composable (() -> Unit)?,
+    shape: Shape,
+    backgroundColor: Color,
+    contentColor: Color,
+    properties: DialogProperties
+) {
+    Dialog(
```

Comment:
After commonization, AlertDialog behaves differently. Check it on `./gradlew run3`:
<img width="775" alt="image" src="https://github.com/JetBrains/compose-multiplatform-core/assets/5963351/7b7e3255-28ab-46c9-8bee-d2763c16af61">
(it shouldn't be this wide, and should have paddings around buttons).

Reply:
`Dialog` issue. I'll fixed in a separate PR

Reply:
> it shouldn't be this wide

It uses `fillMaxWidth` inside common code, so "this wide" looks reasonable except margin and padding.
Also, it behaves differently on Android - it depends on orientation. Investigating how to adopt it

```kt
Dialog(onDismissRequest = {}) {
    Box(Modifier.fillMaxSize().background(Color.Green))
}
```
```kt
AlertDialog(
    onDismissRequest = {},
    confirmButton = { Button(onClick = { }) { Text("Confirm") } },
    text = { Text("Text") },
    backgroundColor = Color.Green
)
```
Dialog | AlertDialog
--- | ---
![Screenshot_20230802_122119](https://github.com/JetBrains/compose-multiplatform-core/assets/1836384/b8f70c1e-84ac-4d05-b938-8a52f72be1b3) | ![Screenshot_20230802_121815](https://github.com/JetBrains/compose-multiplatform-core/assets/1836384/5931d7f1-58a2-405b-ae3d-4db4be793e95)
![Screenshot_20230802_122110](https://github.com/JetBrains/compose-multiplatform-core/assets/1836384/a9df8b24-e889-44f3-ae34-55da3d48a669) | ![Screenshot_20230802_121824](https://github.com/JetBrains/compose-multiplatform-core/assets/1836384/ef883008-edaf-4592-95f1-4e0aa54fd5bd)

Reply:
Sounds good, I checked the default behavior - it adds necessary paddings and doesn't expand to the window

We need to describes this in API changes though